deploy/pulumi/setup-stack.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+set -e
+
+# Usage: ./setup-stack.sh <stack-name> <environment>
+# Example: ./setup-stack.sh gp-api-develop-shadow dev
+
+STACK_NAME=$1
+ENV=$2
+
+if [ -z "$STACK_NAME" ] || [ -z "$ENV" ]; then
+  echo "Usage: ./setup-stack.sh <stack-name> <environment>"
+  echo "Example: ./setup-stack.sh gp-api-develop-shadow dev"
+  exit 1
+fi
+
+# Default values (can be overridden by env vars)
+REGION=${AWS_REGION:-"us-west-2"}
+STATE_BUCKET="s3://gp-api-pulumi-state"
```

Comment:
I think it'd be simpler to use a single state bucket for all of our repos, esp. since we are possibly considering moving to a mono-repo. It reduces the infrastructure surface, and makes it easier to adopt pulumi in other repos (if we don't migrate to monorepo). Could we recreate this bucket as e.g. `goodparty-state`?

Within that global bucket, we should use "projects" to differentiate between repos and "stacks" to differentiate between environments (until we migrate to per-account environments).

The "project" is defined in the `Pulumi.yaml` `name` field, and you're already specifying `stack` via `STACK_NAME` here.